pkg/snapshot/snapshotter/snapshotter.go

```diff
@@ -344,9 +344,8 @@ func (ssr *Snapshotter) takeFullSnapshot(isFinal bool) (*brtypes.Snapshot, error
 		}
 	}
 	lastRevision := resp.Header.Revision
-
-	if ssr.PrevSnapshot.Kind == brtypes.SnapshotKindFull && ssr.PrevSnapshot.LastRevision == lastRevision && ssr.PrevSnapshot.IsFinal == isFinal {
-		ssr.logger.Infof("There are no updates since last snapshot, skipping full snapshot.")
+	if ssr.PrevSnapshot.IsFinal == true {
```

Comment:
```suggestion
	if ssr.PrevSnapshot.IsFinal == true && isFinal == true {
```

This checks whether the previous full snapshot is a final full snapshot, AND whether the final-ness of the previous full snapshot needs to be respected or ignored. In this context, I would also recommend that you rename the variable `isFinal` to a more meaningful name, like `checkPreviousFinalFullSnapshot`.

There are certain cases where we might not want to block a full snapshot because there's a final full snapshot already present in the bucket - this is because "final" full snapshot doesn't really strictly mean final, but rather a temporary final state. For instance, during Gardener CPM, a final full snapshot is taken on the old seed before stopping the etcd, and then the bucket is copied over to the new seed and the etcd is started again. At this point, even though there's a final full snapshot present in the bucket, the new etcdbr agent should continue taking new snapshots on top of the final full snapshot.

@ishan16696 please correct me if I'm wrong.

Reply:
> This checks whether the previous full snapshot is a final full snapshot, AND whether the final-ness of the previous full snapshot needs to be respected or ignored. In this context, I would also recommend that you rename the variable isFinal to a more meaningful name, like checkPreviousFinalFullSnapshot

boolean `isFinal` is not here to indicate about previous final full snapshot. This check of previous full snapshot was final or not is indicated by `ssr.PrevSnapshot.IsFinal`, so this flag `isFinal` indicates wether this current request to trigger out-of-schedule full snapshot is for `final full snapshot` or not. 

@shreyas-s-rao your concern is valid, I have updated the condition:
```go
	if isFinal && ssr.PrevSnapshot.Kind == brtypes.SnapshotKindFull && ssr.PrevSnapshot.LastRevision == lastRevision && ssr.PrevSnapshot.IsFinal {
		ssr.logger.Infof("There are no updates since last snapshot, skipping full snapshot.")
	}
````

@shreyas-s-rao @Shreyas-s14 please take a look.

Reply:
@ishan16696 thanks. Why do we still need to check `ssr.PrevSnapshot.Kind == brtypes.SnapshotKindFull && ssr.PrevSnapshot.LastRevision == lastRevision`? We no longer want to care about the revision from the previous full snapshot right?

Reply:
```go
	if isFinal && ssr.PrevSnapshot.IsFinal && ssr.PrevSnapshot.Kind == brtypes.SnapshotKindFull && ssr.PrevSnapshot.LastRevision == lastRevision  {
		ssr.logger.Infof("There are no updates since final last snapshot, skipping final full snapshot.")
	}
```

as discussed offline with @shreyas-s-rao @Shreyas-s14 this above condition will cover all of backup-restore requirements as well as for CPM.